Student Resources/2 - Introduction to UIKit/4 - Classes and Inheritance/lab/Lab - Classes.playground/Pages/4. Exercise - Class Memberwise Initializers and References.xcplaygroundpage/Contents.swift

```diff
@@ -50,27 +62,32 @@ class ShieldedShip: Fighter {
  
  Then create an instance of `Spaceship` below called `falcon`. Use the memberwise initializer you just created. The ship's name should be "Falcon."
  */
-
+let falcon = Spaceship(name: "Falcon", health: 100, position: 0)
 
 /*:
  Writing initializers for subclasses can get tricky. Your initializer needs to not only set the properties declared on the subclass, but also set all of the uninitialized properties on classes that it inherits from. Go to the declaration of `Fighter` and write an initializer that takes an argument for each property on `Fighter` and for each property on `Spaceship`. Set the properties accordingly. (Hint: you can call through to a superclass' initializer with `super.init` *after* you initialize all of the properties on the subclass).
  
  Then create an instance of `Fighter` below called `destroyer`. Use the memberwise initializer you just created. The ship's name should be "Destroyer."
  */
-
+let destroyer = Fighter(weapon: "lasers", remainingFirePower: 100, name: "Destroyer", health: 100, position: 0)
 
 /*:
  Now go add an initializer to `ShieldedShip` that takes an argument for each property on `ShieldedShip`, `Fighter`, and `Spaceship`, and sets the properties accordingly. Remember that you can call through to the initializer on `Fighter` using `super.init`.
  
  Then create an instance of `ShieldedShip` below called `defender`. Use the memberwise initializer you just created. The ship's name should be "Defender."
  */
-
+let defender = ShieldedShip(shieldStrength: 100, weapon: "cannon", remainingFirePower: 25, name: "shieldedship", health: 100, position: 0)
 
 /*:
  Create a new instance of `Spaceship` called `sameShip` and set it equal to `falcon`. Print out the position of `sameShip` and `falcon`, then call `moveLeft()` on `sameShip` and print out the position of `sameShip` and `falcon` again. Did both positions change? Why? If both were structs instead of classes, would it be the same? Why or why not? Provide your answer in a comment or print statement below.
  */
-
-
+let sameShip = falcon
+print(falcon.position)
+print(sameShip.position)
+sameShip.moveLeft()
+print(sameShip.position)
+print(falcon.position)
+//both positions changed because sameShip is memorized as the same object.
 /*:
```

Comment:
Hey Nathan, the code looks good. This answer is partially correct, but also incomplete. The question asks:
`Did both change, and why?`
`If they were both structs would the positions both have changed, and why or why not?`
If you can answer the questions more fully I can approve the work for this lab.